src/lib/shmem/src/allocator.rs

```diff
@@ -210,62 +228,54 @@ pub struct ShMemBlockSerialized {
 
 unsafe impl Pod for ShMemBlockSerialized {}
 
-impl ShMemBlockSerialized {
-    pub fn to_string_buf(self) -> StringBuf {
-        let mut retval = StringBuf([0; STRING_BUF_NBYTES]);
-
-        let mut offset_buf = [0u8; 32];
-        self.internal.offset.numtoa(10, &mut offset_buf);
-
-        let i1 = offset_buf.iter().filter(|x| **x != 0);
-        let i2 = ";".as_bytes().iter();
-        let i3 = self.internal.chunk_name.iter();
-
-        let i_all = i1.chain(i2).chain(i3);
-
-        retval.0.iter_mut().zip(i_all).for_each(|(x, y)| *x = *y);
-
-        retval
+impl core::fmt::Display for ShMemBlockSerialized {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        let s =
+            core::str::from_utf8(crate::util::trim_null_bytes(&self.internal.chunk_name).unwrap())
+                .unwrap();
+        write!(f, "{};{}", self.internal.offset, s)
     }
+}
 
-    pub fn from_string_buf(string_buf: StringBuf) -> Self {
-        const DELIM: u8 = 59; // Decimal value of ;
+impl core::str::FromStr for ShMemBlockSerialized {
+    type Err = anyhow::Error;
 
-        let lhs_itr = string_buf.0.iter();
-        let mut rhs_itr = string_buf.0.iter();
-        rhs_itr.find(|x| **x == DELIM);
+    // Required method
+    fn from_str(s: &str) -> anyhow::Result<Self> {
+        use core::fmt::Write;
+        use formatting_nostd::FormatBuffer;
 
-        let offset_itr = lhs_itr.take_while(|x| **x != DELIM);
+        if let Some(split_point) = s.find(';') {
+            let (offset_str, path_str) = s.split_at(split_point);
 
-        let mut offset_buf = [0u8; 32];
-        offset_buf
-            .iter_mut()
-            .zip(offset_itr)
-            .for_each(|(x, y)| *x = *y);
+            // let offset = offset_str.parse::<isize>().map_err(Err).unwrap();
+            let offset = offset_str
+                .parse::<isize>()
+                .map_err(Err::<(), core::num::ParseIntError>)
+                .unwrap();
 
-        let mut path_buf: crate::shmalloc_impl::PathBuf =
-            [0; crate::shmalloc_impl::PATH_BUF_NBYTES];
+            let mut chunk_format = FormatBuffer::<{ crate::util::PATH_MAX_NBYTES }>::new();
 
-        path_buf.iter_mut().zip(rhs_itr).for_each(|(x, y)| *x = *y);
+            write!(&mut chunk_format, "{}", &path_str[1..]).unwrap();
```

Comment:
This will panic if `path_str` is 0-length; e.g. if the original string started with `;`. It's also a bit unclear why the first character is being ignored. If the intent is to strip a leading '/', then maybe:

```
let Some(stripped_path_str) = path_str.strip_prefix('/') else {
  return Err(anyhow::anyhow!("missing leading /"))
};
```